src/main/kotlin/dev/isxander/modstitch/base/extensions/Modstitch.kt

```diff
@@ -105,8 +105,14 @@ interface ModstitchExtension {
 
     /**
      * The Java version to target.
+     *
+     * - Defaults to `21` if [minecraftVersion] >= `1.20.5`.
+     * - Defaults to `17` if [minecraftVersion] >= `1.18`.
+     * - Defaults to `16` if [minecraftVersion] >= `1.17`.
+     * - Defaults to  `8` if [minecraftVersion] <= `1.16.5`.
+     * - Has no default value if [minecraftVersion] denotes a snapshot version.
      */
-    val javaTarget: Property<Int>
+    val javaVersion: Property<Int>
```

Comment:
I'd like your thoughts on implementing a deprecated property called `javaTarget` that throws a compiler error when used. Doesn't need to be wired up as a convention or anything. Just having a helpful error would be nice.

Reply:
So, this change is opinionated, and I went with it only because Modstitch is still in its 0.x phase, where breaking changes are somewhat expected, and the user base isn't large enough yet for anybody to care about it. However, if backward compatibility is a concern for you, I can revert the name change instead of introducing a dead property that would merely shift a design-time error to runtime.